geps/gep-3779/index.md

```diff
@@ -42,20 +37,429 @@ Kubernetes core provides NetworkPolicies as one way to do it. Network Policies h
+// objects.
+//
+// Currently, we only support label selectors when targeting Pods.
+// This restriction is intentional to limit the complexity and potential
+// ambiguity of supporting label selectors for arbitrary Kubernetes kinds.
+// Unless there is a very strong justification in the future, we plan to keep this
+// functionality limited to selecting Pods only.
+//
+// This is currently experimental in the Gateway API and should only be used
+// for policies implemented within Gateway API. It is currently not intended for general-purpose
+// use outside of Gateway API resources.
+// +kubebuilder:validation:XValidation:rule="!(has(self.selector)) || (self.kind == 'Pod' && (self.group == '' || self.group == 'core'))",message="Selector may only be set when targeting Pods."
+type PolicyTargetReferenceWithLabelSelectors struct {
+  // Group is the group of the target object.
+  Group Group `json:"group"`
+
+  // Kind is the kind of the target object.
+  Kind Kind `json:"kind"`
+
+  // Selector is the label selector of target objects of the specified kind.
+  Selector *metav1.LabelSelector `json:"selector"`
+}
+
+```
+
+##### Enhanced Discoverability with `gwctl`
+
+A key challenge with `LabelSelector` is the loss of discoverability. It’s easier to see which policies target a `Service` but difficult to determine which policies might affect a specific pod.
+
+To address this, **investment in tooling is required.** Specifically, the `gwctl` CLI tool should be enhanced to provide insights such as:
+
+```sh
+gwctl describe pods pod1 
+
+...
+InheritedPolicies:
+  Type                   Name                                 Target Kind   Target Name/Expression
+  ----                   ----                                 -----------   -----------
+  AuthorizationPolicy  demo-authz-policy-on-payment-pods      Pod           Selector={foo: a, bar: 2}
+...
+
+# List all AuthorizationPolicy resources in json format in the active namespace
+gwctl get authorizationPolicy -o json
+
+```
+
+Without dedicated tooling, the `LabelSelector` approach could significantly degrade the user experience and observability.
+
+##### Targeting All **Pods** in a Namespace
+
+A common case for authorization is to target all the workloads in the namespace (where the policy resource lives). We can achieve the above with a few patterns:
+
+###### Option 1
+
+Target a namespace, `name` MUST be empty (will be the namespace where the policy resource lives).
+
+```yaml
+targetRefs:
+- Kind: Namespace
+```
+
+This however gives the (confusing) impression that Gateways (whether in-cluster or off-cluster gateways) are targeted as well. Beyond the question of "does it really make sense for an E/W policy to be applicable to a N/S Gateway?" it leaves no way for users to do a namespace wide policy that excludes Gateways (no way to exclude Off-cluster Gateways with labels -- see option 3 for alternative).
```

Comment:
Ref:

> To avoid ambiguity in the interpretation of the targets, **policy designs MUST clearly define the extent of the effects of the policy respectively to the object kinds they can target** (semantics of scoping a policy). This can be done via documentation and it typically refers to a known hierarchy of resource kinds.

https://gateway-api.sigs.k8s.io/geps/gep-713/#declared-targets-versus-effective-targets